### PR TITLE
KF-2263 resoruce dispatcher Rock

### PR DESCRIPTION
**Pull request by misohu**
First please review this PR: https://github.com/canonical/resource-dispatcher-rock/pull/2 to keep small PRs.

- rock refactor 
- add non root user 
- add docs with example for testing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,57 +1,52 @@
...
-                "user.kubeflow.org/enabled": "true"
-            }
-        }
-    },
-    "children": {
-        "Secret.v1": [],
-    }
-}
+sudo snap install rockcraft --classic --edge
+sudo snap install skopeo --edge --devmode
```
Comment (Abuelodelanada) [inline README.md]: Should we explicitly ask for `skopeo` installation?

When you install `rockcraft`, `skopeo` is included in the package.


```shell
ubuntu@docker:~$ snap list
Name             Version        Rev    Tracking       Publisher   Notes
bare             1.0            5      latest/stable  canonical✓  base
core18           20230426       2745   latest/stable  canonical✓  base
core20           20230404       1879   latest/stable  canonical✓  base
lxd              5.0.2-838e1b2  24322  5.0/stable/…   canonical✓  -
multipass-sshfs  1.1            145    latest/stable  canonical✓  -
rockcraft        0+git.e4acb2a  737    latest/edge    sergiusens  classic
snapd            2.57.5         17576  latest/stable  canonical✓  snapd
yq               v4.33.3        2178   latest/stable  mikefarah   -


ubuntu@docker:~$ whereis skopeo
skopeo: /usr/bin/skopeo /usr/share/man/man1/skopeo.1.gz
```

**Review by Abuelodelanada — APPROVED** [review]

File: rockcraft.yaml [hunk 2]
```diff
@@ -22,16 +27,23 @@ parts:
...
+    override-build: |
+      craftctl default
+      cp -a resource_dispatcher/. $CRAFT_PART_INSTALL
+  
+  non-root-user:
+    plugin: nil
+    after: [ubuntu]
+    overlay-script: |
+      # Create a user in the $CRAFT_OVERLAY chroot
+      groupadd -R $CRAFT_OVERLAY -g 1000 ubuntu
```
Comment (i-chvets) [inline rockcraft.yaml]: Use `1001` instead of `1000`

File: rockcraft.yaml [hunk 3]
```diff
@@ -22,16 +27,23 @@ parts:
...
-
-  resources:
-    source: resource_dispatcher
-    plugin: dump
-    organize:
-      "*": "app/"
+    override-build: |
+      craftctl default
+      cp -a resource_dispatcher/. $CRAFT_PART_INSTALL
+
```
Comment (i-chvets) [inline rockcraft.yaml:40]: Add security requirements.

**Review by i-chvets — APPROVED** [review]
Approved.

**Pull request merged**